client/client-multi/client-c/src/kaa/platform-impl/common/sha.c

```diff
@@ -14,28 +14,26 @@
  * limitations under the License.
  */
 
-#include "sha1.h"
 #include <stdint.h>
 #include <stddef.h>
+#include <mbedtls/sha1.h>
 #include <string.h>
+
 #include "kaa_common.h"
 #include "platform/ext_sha.h"
-#include <platform/sock.h>
-
 
 kaa_error_t ext_calculate_sha_hash(const char *data, size_t data_size, kaa_digest digest)
 {
     KAA_RETURN_IF_NIL(digest, KAA_ERR_BADPARAM);
+    if ((data && !data_size) || (!data && data_size))
```

Comment:
Why to invent XOR here? Have you noticed that code keeps executing without error event if both `data` and `data_size` are 0?
This function meaningless if `data` or `data_size` is 0. Rewrite this shit accordingly, please.